PostServerApi/Repository/AddPostRepository.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Microsoft.Extensions.Configuration;
+using PostServerApi.Model;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+
+namespace PostServerApi.Repository
+{
+    public class AddPostRepository:IAddPostRepository
+    {
+        private readonly IConfiguration _configuration;
+        public AddPostRepository(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+        public async Task<List<Post>> GetPosts()
+        {
+            try
+            {
+                DataTable table = new DataTable();
+                string query = @"
+             select  PostID,PostTittle,DescriptionOfPost, CreatedDate, UpdatedDate,LikesCount, HeartCount from dbo.Post";
+                string sqlDataSource = _configuration.GetConnectionString("DefaultConnection");
+                SqlDataReader myReader;
+                using (SqlConnection myCon = new SqlConnection(sqlDataSource))
```

Comment:
Can you please change this approach for connecting db? use EntityFramework instead and linq queries. We need to follow the client project code approach so that we can easily align with them.

Reply:
I have  change the database approach for connecting the database